### ISSUE TITLE
Assertion failure at lib\Runtime\Library\JavascriptExceptionMetadata.cpp:146

### ISSUE THREAD

**Issue by yuzurihara** [microsoft/ChakraCore#6178]
Hi,

JsGetAndClearExceptionWithMetadata() crashes due to assertion failure
at lib\Runtime\Library\JavascriptExceptionMetadata.cpp:146

reproducible code:
```
int main()
{
  JsRuntimeHandle runtime;
  JsContextRef context;
  JsValueRef result;
  JsErrorCode jserr;
  unsigned currentSourceContext = 0;

  std::wstring script = LR"***(
undefname("領収書");
)***";

  JsCreateRuntime(JsRuntimeAttributeNone, nullptr, &runtime);
  JsCreateContext(runtime, &context);
  JsSetCurrentContext(context);

  jserr = JsRunScript(script.c_str(), currentSourceContext++, L"", &result);
  if(jserr!=0){
    JsValueRef metadata;
    JsGetAndClearExceptionWithMetadata(&metadata); // assertion failure!
  }
  JsSetCurrentContext(JS_INVALID_REFERENCE);
  JsDisposeRuntime(runtime);
  return 0;
}
```

Build: x64 Debug
OS: Win10 64

**Comment by yuzurihara** [comment]
patch for lib/Runtime/Base/LineOffsetCache.cpp

```
@@ -215,22 +215,23 @@ namespace Js
 
     // Tracks a new line offset in the cache.
     void LineOffsetCache::AddLine(Recycler * allocator, charcount_t characterOffset, charcount_t byteOffset)
     {
         LineOffsetCacheList * characterOffsetList = (LineOffsetCacheList *)(LineOffsetCacheReadOnlyList*)this->lineCharacterOffsetCacheList;
-        characterOffsetList->Add(characterOffset);
         LineOffsetCacheList * byteOffsetList = (LineOffsetCacheList *)(LineOffsetCacheReadOnlyList*)this->lineByteOffsetCacheList;
         if (byteOffsetList == nullptr && characterOffset != byteOffset)
         {
             byteOffsetList = RecyclerNew(allocator, LineOffsetCacheList, allocator);
             byteOffsetList->Copy(characterOffsetList);
             this->lineByteOffsetCacheList = byteOffsetList;
+            byteOffsetList->Add(byteOffset);
         }
         else if (byteOffsetList != nullptr)
         {
             byteOffsetList->Add(byteOffset);
         }        
+        characterOffsetList->Add(characterOffset);
 
 #if DBG
         Assert(this->lineByteOffsetCacheList == nullptr || this->lineByteOffsetCacheList->Count() == this->lineCharacterOffsetCacheList->Count());
         if (this->lineCharacterOffsetCacheList->Count() > 1)
         {
```

**Comment by boingoing** [comment]
This was fixed in master via e997d44c0cd5148bfadb12791c17dd7138a183a3. Should be in our next release branch after 1.11.

Can you verify the master branch does not repro?

**Comment by yuzurihara** [comment]
It works fine!
I'll look forward to the next release.

Thank you for your kindness.
